python_modules/dagster/dagster/core/definitions/event_metadata/__init__.py

```diff
@@ -581,17 +628,87 @@ def validate_table(context, df):
         check.inst_param(asset_key, "asset_key", AssetKey)
         return DagsterAssetMetadataEntryData(asset_key)
 
+    @staticmethod
+    def table(records: List[TableRecord], schema: TableSchema) -> "TableMetadataEntryData":
+        """Static constructor for a metadata value wrapping arbitrary tabular data as
+        :py:class:`TableMetadataEntryData`. Can be used as the value type for the `metadata`
+        parameter for supported events. For example:
+
+        .. code-block:: python
+
+            @op
+            def emit_metadata(context):
+                yield ExpectationResult(
+                    success=not has_errors,
+                    label="is_valid",
+                    metadata={
+                        "errors": EventMetadata.table(
+                            records=[
+                                TableRecord(code="invalid-data-type", row=2, col="name"}]
+                            ],
+                            schema=TableSchema(
+                                columns=[
+                                    TableColumn(name="code", type="string"),
+                                    TableColumn(name="row", type="int"),
+                                    TableColumn(name="col", type="string"),
+                                ]
+                            )
+                        ),
+                    },
+                )
 
+        Args:
+            records (List[TableRecord]): The data as a list of records (i.e. rows).
+            schema (TableSchema): A schema for the table.
+        """
+        return TableMetadataEntryData(records, schema)
+
+    @staticmethod
+    def table_schema(
+        schema: TableSchema,
+    ) -> "TableSchemaMetadataEntryData":
+        """Static constructor for a metadata value wrapping a table schema as
+        :py:class:`TableSchemaMetadataEntryData`. Can be used as the value type
+        for the `metadata` parameter for supported events. For example:
+
+        .. code-block:: python
+
+            schema = TableSchema(
+                columns = [
+                    TableColumn(name="id", type="int"),
+                    TableColumn(name="status", type="bool"),
+                ]
+            )
+
+            DagsterType(
+                type_check_fn=some_validation_fn,
+                name='MyTable',
+                metadata_entries=[
```

Comment:
This should be `metadata={...`, right?

Reply:
I don't think so? In adding to metadata to Dagster types, I followed the API of events. If you look in `core/definitions/events.py`, you'll see the various classes can take either`metadata` (`Dict[str, ParseableMetadataEntry]`) or `metadata_entries` (`List[EventMetadataEntry]`). I followed the example of this API.

FWIW I find exposing two different arguments to pass the same info in different format confusing, seems it would be better in the long run to standardize on one? Preferably `metadata_entries`, since it let's you specify a `description` as well.. But I assume the 2 are there for backward compatibility reasons.

Reply:
The method being documented here doesn't have a label argument, but the invocation in the code example in the doc string uses a label argument, so something isn't matching, right?

Reply:
And yes, it's for backwards compatibility reasons that we unfortunately have both.